src/server/replica.h

```diff
@@ -142,6 +156,7 @@ class Replica {
   ::boost::fibers::fiber sync_fb_;
   std::vector<std::unique_ptr<Replica>> shard_flows_;
 
+  std::unique_ptr<base::IoBuf> leftover_buf_;
   std::unique_ptr<facade::RedisParser> parser_;
```

Comment:
For some reason you like unique pointers a lot 🤔 Is it for reducing struct size? Given that IoBuf is not that large I'd use an optional instead

Reply:
I indeed use unique_ptr where the performance impact is not important and where I want to control the lifecycle of the object easily. You can use optional - i do not have strong opinion about the matter.